### PR TITLE
add dependency review

### PR DESCRIPTION
**Pull request by UlisesGascon**
Refs: https://github.com/nodejs/nodejs.org/issues/5659

### CODE REVIEW

**Review by ovflowd — CHANGES_REQUESTED** [review]
We already have GitHub's built-in dependency scanning.

**Review by ovflowd — APPROVED** [review]

File: .github/workflows/dependency-review.yml
Comment (ovflowd) [inline .github/workflows/dependency-review.yml:1]: Should we mark this as a required Pull Request check?

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+on: [pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
```
Comment (ovflowd) [inline .github/workflows/dependency-review.yml:21]: What is a Harden Runner? Could you add a link to docs here?

Comment (UlisesGascon) [reply .github/workflows/dependency-review.yml:21]: Sure! Harden Runner acts as a firewall for Github Actions. Currently, it collects all HTTP requests made within the job. The purpose is to review the reports and create specific rules to allow or block requests to unauthorized domains. You can find more information [here](https://github.com/step-security/harden-runner).

Should I add a proper documentation for this?

**Pull request merged**